### PR TITLE
Add tests for read_graph

### PR DESCRIPTION
**Pull request by Plastova**
Added badarg and correctness tests for read_graph functionality

### CODE REVIEW

File: cpp/oneapi/dal/io/csv/test/read_graph_badarg.cpp [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+            outf << file_content << std::endl;
+        }
+        else {
+            throw invalid_argument(dal::detail::error_messages::file_not_found());
+        }
+        outf.close();
+    }
+
+    void delete_test_data(std::string filename) {
+        std::remove(filename.c_str());
```
[the remark this answers is not in the document]
Comment (Plastova) [reply cpp/oneapi/dal/io/csv/test/read_graph_badarg.cpp:76]: This function removes files https://en.cppreference.com/w/cpp/io/c/remove

**Review by orazve — APPROVED** [review]

**Pull request merged**